src/components/Tokens/TokenDetails/TransactionsTable.tsx

```diff
@@ -0,0 +1,118 @@
+import { Table } from 'components/Table'
+import { LinkCell, TextCell } from 'components/Table/Cells'
+import { useActiveLocale } from 'hooks/useActiveLocale'
+import { useMemo } from 'react'
+import { Column } from 'react-table'
+import { useTheme } from 'styled-components'
+import { shortenAddress } from 'utils/addresses'
+import { NumberType, useFormatter } from 'utils/formatNumbers'
+
+import { mockSwapData } from './mockData'
+import { Swap, SwapAction, SwapInOut } from './types'
+import { getLocaleTimeString, getSwapType } from './utils'
+
+enum ColumnHeader {
+  Time = 'Time',
+  Type = 'Type',
+  Amount = 'Amount',
+  For = 'For',
+  USD = 'USD',
+  Maker = 'Maker',
+}
+
+export function TransactionsTable({ referenceTokenAddress }: { referenceTokenAddress: string }) {
+  const locale = useActiveLocale()
+  const { formatNumber } = useFormatter()
+  const theme = useTheme()
+
+  const columns: Column<Swap>[] = useMemo(() => {
+    const getColor = (action: SwapAction) => (action === SwapAction.Buy ? theme.success : theme.critical)
```

Comment:
does this need to be a fn? it's called 4 times with the same val

Reply:
Looking at the Figma, the colors for txs are only used for the PDP not TDP

Reply:
I've created a reusable `ColorText` component instead. 

There was some back-and-forth regarding the color of the transaction data. For now, it's in the latest mocks, but this may change.